Systems/ArchipelagoSystem.cs

```diff
@@ -104,6 +104,7 @@ public override void OnWorldLoad()
 
             session = new();
             session.session = newSession;
+            session.hardmodeAsItem = config.hardmodeAsItem;
```

Comment:
Instead of having a field in the session, just check the config when Hardmode is received. That way, the player can change this setting while in-game (ex they want to delay Hardmode until they get some basic gear, so they change the setting after their first mining trip).